t/lib/TestDB.pm

```diff
@@ -12,11 +12,37 @@ sub new {
 
     open my $fh, '<', 'share/test_data.sql';
 
+    my $continued_line = "";
+    my $last_line_was_continued = 0;
+
     while (<$fh>) {
         $ENV{DBIC_TRACE}
           and warn $_;
-
-        $schema->storage->dbh_do( sub { $_[1]->do($_) } );
+	my $line = $_;
+
+	# Remove comments from SQL-File
+	my $comment_start = index ($line, '--');
+	$line = substr ($line, 0, $comment_start);
+    # Remove trailing and leading whitespaces
+    $line =~ s/^\s+|\s+$//g;
+	next if $line =~ /^\s*$/;
+	my $last_character = substr ($line, -1);
+
+	# Check if last character of a line is not a semicolon, because in this case
+	# the line currently looked at is not a complete SQL-Statement and we can't
+	# execute it via DBICx::TestDatabase and first need to add all following lines
+	# to our statement until we find a semicolon at the end of a line.
+	if ($last_character ne ';') {
```

Comment:
Perl allows you to do this easier:

    if( $line =~ m/ ; $ /x) {

I inserted whitespace and added `/x` again for readability.